### PR TITLE
add no macros example, remove bogus print, fix crash on empty schedule

### PR DESCRIPTION
**Pull request by wucke13**

### CODE REVIEW

File: hypervisor/src/hypervisor/partition.rs [hunk 1]
```diff
@@ -507,7 +507,7 @@ impl Base {
 
 #[derive(Debug)]
 pub(crate) struct Partition {
-    base: Base,
+    pub(crate) base: Base,
```
Comment (dadada) [inline hypervisor/src/hypervisor/partition.rs:510]: This exposes way more than required. Better implement `Partition::name()`.

File: examples/hello_part_no_macros/src/main.rs [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+                    .ok()
+                    .unwrap();
+
+                info!("Received via Sampling Port: {:?}, valid: {valid:?}", data)
+            }
+
+            Hypervisor::periodic_wait().unwrap();
+        }
+    }
+}
```
Comment (dadada) [inline examples/hello_part_no_macros/src/main.rs:153]: Duplicate code. You can however add a second binary to the `hello_part` crate and put the process code in `lib`.

File: examples/hello_part_no_macros/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+    info!("Start Aperiodic");
+    for i in 0..i32::MAX {
+        if let SystemTime::Normal(time) = Hypervisor::get_time() {
+            // round the time to an integer value of milliseconds
+            let round = Duration::from_millis(time.as_millis() as u64);
+            info!("{:?}: AP MSG {i}", format_duration(round).to_string());
+        }
+        std::thread::sleep(Duration::from_millis(1))
+    }
+}
```
Comment (dadada) [inline examples/hello_part_no_macros/src/main.rs:111]: Also duplicate code

File: examples/hello_part_no_macros/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+a653rs = { workspace = true }
+a653rs-postcard = { git = "https://github.com/DLR-FT/a653rs-postcard.git", branch = "main" }
+a653rs-linux = { path = "../../partition" }
+once_cell.workspace = true
+serde = "1.0"
+log = "0"
+humantime = "2.1"
```
Comment (dadada) [inline examples/hello_part_no_macros/Cargo.toml:17]: This should probably be workspace dependencies.

File: examples/hello_part_no_macros/src/main.rs [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+//! ## FAQ
+//!
+//! - What is [`StartContext<Hypervisor>`]?
+//!   - Some parts of the ARINC 653 API are only available on a specific mode.
+//!     In particular, the process and port creationg API can only be called
+//!     during {`COLD`,`WARM`}`_START` mode (see [`OperatingMode`]).
+//!     `StartContext` is a struct that exposes this API only during
+//!     {`COLD`,`WARM`}`_START` mode, and can not be constructed by the user.
+//!     This makes misuse of the API (as in attempting to create a port/process
+//!     during `NORMAL`) impossible.
```
Comment (dadada) [inline examples/hello_part_no_macros/src/main.rs:12]: This should be part of the crate-level documentation and link to it.

File: hypervisor/src/hypervisor/scheduler.rs [hunk 6]
```diff
@@ -80,11 +79,8 @@ impl<'a> PartitionTimeframeScheduler<'a> {
                 let res = self.partition.run_aperiodic_process(self.timeout);
                 if self.handle_partition_result(res)? == Some(false) {
                     // Aperiodic process was also not run
-                    return Err(LeveledError::new(
-                        SystemError::Panic,
-                        ErrorLevel::Partition,
-                        anyhow!("At least one periodic or aperiodic process is expected to exist"),
-                    ));
+                    let part_name = self.partition.base.name();
+                    warn!("partition {part_name}: no process is scheduled")
```
Comment (dadada) [inline hypervisor/src/hypervisor/scheduler.rs:83]: Rather we should check at the end of cold_start if there is no process. If the partition is run multiple times a second, this produces too much log-spam on a relatively high log-level.

**Review by dadada — CHANGES_REQUESTED** [review]

File: examples/hello_part_no_macros/src/main.rs (hunk 2 above)
Comment (wucke13) [reply examples/hello_part_no_macros/src/main.rs:153]: I strongly oppose this. The idea of examples is to be relatively easy, independent snippets of code that can be easily copied around. I myself get confused quite easily when crate examples are too mixed up with each other, sharing the same code factored out in a dedicated library. I think having no coupling, canonical single-file examples is the best way to go, and DRY should not be a decision driver for examples. On mostly any other kind of code I would agree with you, just not for examples.

File: examples/hello_part_no_macros/src/main.rs (hunk 5 above)
Comment (wucke13) [reply examples/hello_part_no_macros/src/main.rs:12]: Yes, actually, this should be part of the `a653rs` documentation.

File: hypervisor/src/hypervisor/scheduler.rs (hunk 6 above)
Comment (wucke13) [reply hypervisor/src/hypervisor/scheduler.rs:83]: This then would be a check inside the partition, AFAICT. I'm not an expert on this however.

File: hypervisor/src/hypervisor/partition.rs (hunk 1 above)
Comment (wucke13) [reply hypervisor/src/hypervisor/partition.rs:510]: Agreed! Will do implement your suggestion

File: examples/hello_part_no_macros/src/main.rs (hunk 2 above)
Comment (dadada) [reply examples/hello_part_no_macros/src/main.rs:153]: It is still double the effort for maintaining the example. :-/ Maybe drop the hello_part altogether. We have other examples for the macros like the ping_client and ping_server.

File: examples/hello_part_no_macros/Cargo.toml (hunk 4 above)
Comment (dadada) [reply examples/hello_part_no_macros/Cargo.toml:17]: Yeah ok for the example it is probably easier to copy-paste. But then again it should *all* not be workspace dependencies. Pick one?

File: examples/hello_part_no_macros/src/main.rs (hunk 2 above)
Comment (sevenautumns) [reply examples/hello_part_no_macros/src/main.rs:153]: I agree with @wucke13 but I also think that the hello world example is way to large for a hello world example.
Maybe we should prune some unnecessary complexity from it

File: hypervisor/src/hypervisor/partition.rs (hunk 1 above)
Comment (wucke13) [reply hypervisor/src/hypervisor/partition.rs:510]: Done

File: examples/hello_part_no_macros/src/main.rs (hunk 3 above)
Comment (wucke13) [reply examples/hello_part_no_macros/src/main.rs:111]: Intentional

File: hypervisor/src/hypervisor/scheduler.rs (hunk 6 above)
Comment (wucke13) [reply hypervisor/src/hypervisor/scheduler.rs:83]: I thought further about this. I think it is completely fine to spam the log if there is a partition that at runtime creates an invalid configurration (no processes...). If anything, this should grab the attention of the developer, so spamming is justified. We could more clever logging (akin "20 identical messages dropped") but I don't see the justification for making this, considering that this only ever occurs when there is a low-key screw up by the developer to begin with.

File: examples/hello_part_no_macros/Cargo.toml (hunk 4 above)
Comment (wucke13) [reply examples/hello_part_no_macros/Cargo.toml:17]: I agree. I just copied what was already there. I'm in favor of adjusting this, however, I think we should prepare an extra PR for that.

**Pull request merged**